app/src/main/java/com/ciandt/dragonfly/example/features/download/DownloadReceiver.kt

```diff
@@ -0,0 +1,25 @@
+package com.ciandt.dragonfly.example.features.download
+
+import android.app.DownloadManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+
+class DownloadReceiver : BroadcastReceiver() {
+
+    override fun onReceive(context: Context, intent: Intent) {
+
+        val id = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, 0)
+        if (id !in DownloadHelper.getDownloads(context)) {
```

Comment:
Think it would be better to have a DownloadHelper.exists(id) or DownloadHelper.isValid(id), no?

Or getDownloadedFile() could return a null object or throw an exception?